### PR TITLE
network: fix failures in updating MAC address

### PR DESCRIPTION
**Pull request by yuwata**
Maybe related to #16105.

cc @ddstreet and @mrc0mmand.

### CODE REVIEW

**Comment by yuwata** [comment]
See https://ci.centos.org/job/systemd-pr-build-vagrant-sanitizers/6873/artifact//systemd-centos-ci/artifacts_all/artifacts_HtvfM0/vagrant-logs.AjZ/vagrant-arch-sanitizers-clang-testsuite.GDS/systemd-networkd_sanitizers_PASS.log

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -3932,78 +3932,114 @@ int link_update(Link *link, sd_netlink_message *m) {
...
                                 if (r < 0)
-                                        return log_link_warning_errno(link, r, "Could not update MAC for ndisc: %m");
+                                        return log_link_warning_errno(link, r, "Could not restart Router Advertisement: %m");
                         }
                 }
+
+                if (link->ndisc) {
+                        r = sd_ndisc_set_mac(link->ndisc, &link->mac);
+                        if (r < 0)
+                                return log_link_warning_errno(link, r, "Could not update MAC for ndisc: %m");
```
Comment (keszybz) [inline src/network/networkd-link.c]: "ndisc" → "Ndisc".

**Comment by mrc0mmand** [comment]
The CI looks happy and so do the CI logs, so from my point of view this definitely looks great. I'd wait for @ddstreet for a more educated review than this one though :-)

**Comment by ddstreet** [comment]
Yep this definitely looks correct to me.

Re: #16105 though, I think that was failing when the mac didn't change, so I'm not sure if this will fix the failure in that test.

**Comment by yuwata** [comment]
> Re: #16105 though, I think that was failing when the mac didn't change, so I'm not sure if this will fix the failure in that test.

Yeah, this PR seems not related to the issue #16105, unfortunately. I've re-read the log linked at https://github.com/systemd/systemd/issues/16105#issuecomment-648735236, and the assertion fixed by this PR does not triggered in that case...

Anyway, thank you for the review. Merging this as I got the positive comments.

**Pull request merged**